src/Facility.CodeGen.CSharp/CSharpGenerator.cs

```diff
@@ -444,8 +445,22 @@ private CodeGenFile GenerateHttpMapping(HttpServiceInfo httpServiceInfo, Context
 								code.WriteLine($"new HttpMethodMapping<{requestTypeName}, {responseTypeName}>.Builder");
 								using (code.Block("{", "}.Build();"))
 								{
-									string httpMethodName = CodeGenUtility.Capitalize(httpMethodInfo.Method.ToString().ToLowerInvariant());
-									code.WriteLine($"HttpMethod = HttpMethod.{httpMethodName},");
+									string httpMethod = httpMethodInfo.Method;
+									if (httpMethod == HttpMethod.Get.Method    ||
+									    httpMethod == HttpMethod.Put.Method    ||
+									    httpMethod == HttpMethod.Head.Method   ||
+									    httpMethod == HttpMethod.Post.Method   ||
+									    httpMethod == HttpMethod.Trace.Method  ||
+									    httpMethod == HttpMethod.Delete.Method ||
+									    httpMethod == HttpMethod.Options.Method)
```

Comment:
HEAD, TRACE, and OPTIONS are [not supported](https://github.com/FacilityApi/Facility/blob/b390fcccaf95017d37fb1bacf9d902d1a67426bb/src/Facility.Definition/Http/HttpMethodInfo.cs#L375) so we don't need to check for them here.